### ISSUE TITLE
Rename curves and write out to new file

### ISSUE THREAD

**Issue by PetroDude** [kinverarity1/lasio#154]
I can read an LAS file and write it out using your examples. However, I have tried several different methods to rename a curve. Depending on which method I try, I can get it renamed in memory (according to l.curves), but when I write it to a file, the original name appears.

**Comment by dagrha** [comment]
@PetroDude, can you give a code example?

**Comment by kinverarity1** [comment]
I'm not surprised this doesn't work - there's some odd code in there related to this which allows reading files that have several curves with the same mnemonic. I'll have a look later today hopefully.

---- Dan H. wrote ----


@PetroDude<https://github.com/petrodude>, can you give a code example?

**Comment by PetroDude** [comment]
Here is one method I tried. It seems like it is pretty straightforward.
`import lasio

l = lasio.read('Trash.las')

for curve in l.curves:
    if curve.mnemonic != 'DEPT':
        curve.mnemonic = "New_" + curve.original_mnemonic

with open('Trash2.las', mode='w') as f:
   l.write(f, wrap=False)`

The 'Trash2.las' file still contains the original curve names.

After I run the above code and then run l.curves, I get something like the following:

[CurveItem(mnemonic=DEPT, unit=F, value=00 000 00 00, descr=Depth (MD), original_mnemonic=DEPT, data.shape=(16939,)),
 CurveItem(mnemonic=New_BADDEN, unit=, value=00 001 00 00, descr=, original_mnemonic=BADDEN, data.shape=(16939,)),
 CurveItem(mnemonic=New_BHDEN, unit=, value=00 002 00 00, descr=, original_mnemonic=BHDEN, data.shape=(16939,)),
 CurveItem(mnemonic=New_BVH, unit=dec, value=00 003 00 00, descr=, original_mnemonic=BVH, data.shape=(16939,)),

So the mnemonic is getting changed as I would expect. The new mnemonic name is just not being changed.

**Comment by dagrha** [comment]
I tried to fix this by changing [this line](https://github.com/kinverarity1/lasio/blob/master/lasio/writer.py#L107) in `writer.py` from 
```
for header_item in las.curves:
    mnemonic = header_item.original_mnemonic
```
to
```
for header_item in las.curves:
    mnemonic = header_item.mnemonic
```
but that did not work.

**Comment by dagrha** [comment]
However, if I also change the lines in [`get_formatter_function`](https://github.com/kinverarity1/lasio/blob/master/lasio/writer.py#L170) that say `.original_mnemonic` to simply `.mnemonic`, the new curves _do_ get written to the LAS file. 

Would it make sense to add an argument to the `write` method, something like `use_new_mnemonics=False`, where if True it uses `.mnemonic` rather than `.original_mnemonic`?

**Comment by PetroDude** [comment]
I really can't see much reason why you would ever want to write the original mnemonic. Since you have to explicitly change mnemonic to something other than original_mnemonic, I think the default should be ` .mnemonic`.

**Comment by kinverarity1** [comment]
Thanks for the discussion and examples, I dont have a chance to test code at the moment but will look at it when I can.

The reason for this stuff is to deal with the edge case - not all that uncommon - of  duplicate mnemonics. The code is designed to write back out the original mnemonic by default. This is important, so that automated use of lasio doesnt go changing curve mnemonics. We need to distinguish between changes made in reading the file and adding data, and changes made by the user.
________________________________

**Comment by kinverarity1** [comment]
OK should be fixed in master.

Thanks for the example code @PetroDude, I've added a unit test with your code (tests/test_rename_curve.py) and it's passing Travis CI for all Python distributions.

Thanks for the suggestions @dagrha - I chose instead to retain the behaviour I mentioned above for dealing with duplicate and missing mnemonics by moving that behaviour to the ``HeaderItem`` method ``HeaderItem.set_session_mnemonic_only(value)``.
